runtime/interpreter/value.go

```diff
@@ -1868,18 +1898,19 @@ func (v *ArrayValue) iterate(
 		}
 	}
 
-	if v.IsResourceKinded(interpreter) {
-		interpreter.withMutationPrevention(v.StorageID(), iterate)
-	} else {
-		iterate()
-	}
+	interpreter.withMutationPrevention(v.StorageID(), iterate)
 }
 
 func (v *ArrayValue) Walk(interpreter *Interpreter, walkChild func(Value)) {
-	v.Iterate(interpreter, func(element Value) (resume bool) {
-		walkChild(element)
-		return true
-	})
+	v.Iterate(
+		interpreter,
+		func(element Value) (resume bool) {
+			walkChild(element)
+			return true
+		},
+		// TODO: Not supposed to panic with container mutation error.
```

Comment:
Adding locationRange to these two methods explodes in the code, so I left it to make the changes minimal.
Both here and below, the two functions are not called from user code. Instead these are being used internally for other validations. Hence are not supposed to panic with container mutation errors.

Reply:
Yeah, let's do that in a separate PR